kernel/vmvalues.ml

```diff
@@ -372,9 +372,9 @@ let rec whd_accu a stk =
       | _           -> assert false
       end
   | i when Int.equal i Obj.custom_tag ->
-    Vint64 (Obj.magic i)
+    Vint64 (Obj.magic at)
```

Comment:
Is this code even alive?
- With 64 bit ocaml ints are ocaml ints and get handled at https://github.com/coq/coq/blob/6b5ff65feda89087f80101e1fdb5f5c9af615919/kernel/vmvalues.ml#L399 producing Vconstr_const
- With 32 bit ocaml ints are Int64.t, they have custom_tag not closure_tag so they are handled at https://github.com/coq/coq/blob/6b5ff65feda89087f80101e1fdb5f5c9af615919/kernel/vmvalues.ml#L414 producing Vint64 and not here in whd_accu

Floats probably have a similar story, at least I doubt they have closure_tag.

Reply:
Note that this code is executed on the **argument** of a thing with closure tag (i.e., an accumulator), not on the thing itself. But  you might be right that accumulators never have an `int64` or a `float` as direct argument (though they might be applied to them, but that is a different code path). I will change the code to `assert false`.